src/extension.ts

```diff
@@ -115,59 +117,63 @@ export function activate(ctx: vscode.ExtensionContext): void {
       }
     )
   );
+
+  async function pickTest(isBenchmark: boolean) {
+    const tests = isBenchmark ? await getMesonBenchmarks(buildDir) : await getMesonTests(buildDir)
+
+    const items = [
+      {
+        label: "all",
+        detail: `Run all ${isBenchmark ? "benchmarks" : "tests"}`,
+        description: "(meta-target)",
+        picked: true
+      },
+      ...tests.map<vscode.QuickPickItem>(t => ({
+        label: t.name,
+        detail: `Timeout: ${t.timeout}s, ${
+          !isBenchmark && t.is_parallel ? "run in parallel" : "run serially"
+        }`,
+        description: t.suite.join(","),
+        picked: false
+      }))
+    ];
+
+    const result = await vscode.window.showQuickPick(items);
+    return result?.label;
+  }
+
+  async function runTests(name: string, isBenchmark: boolean) {
+    await runMesonTests(
+      workspaceRelative(extensionConfiguration("buildFolder")),
+      isBenchmark,
+      name == "all" ? null : name
+    );
+  }
+
+  async function runTestsOrBenchmarks(isBenchmark: boolean, name?: string) {
+    name ??= await pickTest(isBenchmark);
+
+    if (name != null) {
```

Comment:
Shouldn't this be !== ?

Reply:
`!= null` is OK in this case as `name` will be undefined if the pick is cancelled, or "all" if the test all is selected. It could be `!== undefined`, but whilst I'm far from a Typscript guru some that I worked with preferred `!= null` when I queried the same thing. I guess it's more defensive, which seems fair.

The run build fix the other day had to be `!== null` as the code there was using null for pick cancelled and undefined for build all, so selecting all was just being treated like pick cancelled. It could probably do with some refactoring to normalise between picking a build target and picking a test :)

Reply:
Added another commit to the PR: aab22834b5d4cdfd9100aa5758ab5f90ecbf653b

Hopefully makes the logic a bit clearer and similar across build/test.

I notice it uses `createQuickPick` for build, presumably so it can set the busy flag whilst resolving the available build targets. 

Test/benchmarks just use `showQuickPick`. 

They should maybe be the same? Not really sure the busy flag makes sense as I think the resolution must be pretty quick as configure must have been done for it to work at all.

Indeed, if configure hasn't been done, the pick will just sit there, so that needs a bit of polish too.

Reply:
That makes two of us who are not typescript gurus :) you're explanation makes sense to me, and I guess typescript should be able to catch the case of undefined anyway